test/fft_utest.cpp

```diff
@@ -483,3 +549,244 @@ TYPED_TEST(FftTest, TestFftGf2n) // NOLINT
+{
+    std::vector<uint64_t> qs({7681, 12289, 65537});
+    std::vector<uint64_t> ns({256, 512, 1024});
+
+    for (std::vector<uint64_t>::iterator it = qs.begin(); it != qs.end();
+         it++) {
+        TypeParam q = *it;
+
+        for (std::vector<uint64_t>::iterator it2 = ns.begin(); it2 != ns.end();
+             it2++) {
+            TypeParam n = *it2;
+
+            ASSERT_EQ(quadiron::arith::is_prime(q), true);
+
+            std::cout << "choosing q=" << q << " n=" << n << "\n";
+
+            std::unique_ptr<gf::Field<TypeParam>> gf = nullptr;
+            std::unique_ptr<quadiron::vec::Vector<TypeParam>> a = nullptr;
+            std::unique_ptr<quadiron::vec::Vector<TypeParam>> _a1 = nullptr;
+            std::unique_ptr<quadiron::vec::Vector<TypeParam>> _a2 = nullptr;
+            std::unique_ptr<quadiron::vec::Vector<TypeParam>> _a3 = nullptr;
+            std::unique_ptr<fft::FourierTransform<TypeParam>> fft = nullptr;
+            std::unique_ptr<fft::FourierTransform<TypeParam>> fft2 = nullptr;
+            std::unique_ptr<fft::FourierTransform<TypeParam>> fft3 = nullptr;
+
+            gf = gf::alloc<gf::Field<TypeParam>, gf::Prime<TypeParam>>(q);
+
+            // TypeParam p = gf->get_primitive_root();
+            // std::cout << "p=" << p << "\n";
+            TypeParam r = gf->get_nth_root(n);
+            // std::cout << "r=" << r << "\n";
+            int l = arith::log2<TypeParam>(n);
+            fft = std::make_unique<fft::Naive<TypeParam>>(*gf, n, r);
+            try {
+                fft2 = std::make_unique<fft::Large<TypeParam>>(*gf, l, r);
+            } catch (...) {
+                fft2 = nullptr;
+            }
+            try {
+                fft3 =
+                    std::make_unique<fft::Radix2<TypeParam>>(*gf, n, n, 1024);
+            } catch (...) {
+                fft3 = nullptr;
+            }
+
+            a = std::make_unique<quadiron::vec::Vector<TypeParam>>(*gf, n);
+            _a1 = std::make_unique<quadiron::vec::Vector<TypeParam>>(*gf, n);
+            _a2 = std::make_unique<quadiron::vec::Vector<TypeParam>>(*gf, n);
+            _a3 = std::make_unique<quadiron::vec::Vector<TypeParam>>(*gf, n);
+            a->rand();
+
+            uint64_t start1 = quadiron::hw_timer();
+            fft->fft(*_a1, *a);
+            uint64_t end1 = quadiron::hw_timer();
+
+            uint64_t start2 = quadiron::hw_timer();
+            if (fft2) {
+                fft2->fft(*_a2, *a);
+            }
+            uint64_t end2 = quadiron::hw_timer();
+
+            uint64_t start3 = quadiron::hw_timer();
+            if (fft3) {
+                fft3->fft(*_a3, *a);
+            }
+            uint64_t end3 = quadiron::hw_timer();
+
+            if (fft2) {
+                ASSERT_EQ(*_a1 == *_a2, true);
+            }
+            if (fft3) {
+                ASSERT_EQ(*_a1 == *_a3, true);
+            }
+
+            std::cout << "naive: " << end1 - start1;
```

Comment:
removes these `std::cout`: test are not supposed to produce output (and I would even say that unit test are not a good place for benchmarking)